node_io/test/directory_test.dart

```diff
@@ -15,10 +16,87 @@ void main() {
       expect(Directory.current.path, process.cwd());
     });
 
-    test('exists', () {
+    test('exists', () async {
       expect(Directory.current.existsSync(), isTrue);
       expect(Directory.current.exists(), completion(isTrue));
       expect(Directory.current.isAbsolute, isTrue);
+
+      expect(
+          await new Directory(join(Directory.current.path, "__dummy__"))
```

Comment:
nit: I'd maybe add a helper to create a Directory and DRY up these tests a bit. E.g.

```dart
Directory dir(String name) {
  return new Directory(join(Directory.current.path, name));
}

// In tests:
expect(dir('__dummy__').exists(), completion(isFalse));
final stat = await dir('__dummy__').stat();
expect(stat.type, FileSystemEntityType.notFound);
```

Reply:
ok